codegen/lib/objc-codegen.js

```diff
@@ -0,0 +1,280 @@
+var fs = require('fs');
+var ejs = require('ejs');
+
+/**
+ * Generate iOS Client-side Objective-C representation of the models.
+ *
+ * @param {Object} app The loopback application created via `app = loopback()`.
+ * @returns {string} An array of the generated source file contents.
+ */
+exports.objcModels = function generateServices(app, modelPrefix) {
+
+  var models = describeModels(app);
+
+  addObjCNames(models, modelPrefix);
+
+  var objcModelHTemplate = readTemplate('./objc-model-h.ejs');
+  var objcModelMTemplate = readTemplate('./objc-model-m.ejs');
+  var objcRepoHTemplate  = readTemplate('./objc-repo-h.ejs');
+  var objcRepoMTemplate  = readTemplate('./objc-repo-m.ejs');
+
+  var ret = {};
+
+  for (var modelName in models) {
+    var script = renderContent(objcModelHTemplate, models[modelName]);
+    ret[models[modelName].objcModelName + '.h'] = script;
+
+    var script = renderContent(objcModelMTemplate, models[modelName]);
+    ret[models[modelName].objcModelName + '.m'] = script;
+
+    var script = renderContent(objcRepoHTemplate, models[modelName]);
+    ret[models[modelName].objcModelName + 'Repository.h'] = script;
+
+    var script = renderContent(objcRepoMTemplate, models[modelName]);
+    ret[models[modelName].objcModelName + 'Repository.m'] = script;
+  }
+
+  return ret;
+};
+
+function describeModels(app) {
+  var result = {};
+  for(var model in app.models) {
+    model.get;
+  }
+  app.handler('rest').adapter.getClasses().forEach(function(c) {
+    var name = c.name;
+
+    if (!c.ctor) {
+      // Skip classes that don't have a shared ctor
+      // as they are not LoopBack models
+      console.error('Skipping %j as it is not a LoopBack model', name);
+      return;
+    }
+    c.methods.forEach(function fixArgsOfPrototypeMethods(method) {
+      var ctor = method.restClass.ctor;
+      if (!ctor || method.sharedMethod.isStatic) return;
+      method.accepts = ctor.accepts.concat(method.accepts);
+    });
+    c.pluralName = c.sharedClass.ctor.pluralModelName;
+    c.params =  app.models[c.name].definition.properties;
+    c.baseModel = app.models[c.name].definition.settings.base;
+
+    if (c.baseModel != null && typeof(c.baseModel) === "function") {
+      c.baseModel = "";
+    }
+    if (app.models[c.name].definition._ids != null) {
+      c.isGenerated = app.models[c.name].definition._ids[0].property.generated;
+    } else {
+      c.isGenerated = false;
+    }
+    c.relations = app.models[c.name].definition.settings.relations;
+    c.acls = app.models[c.name].definition.settings.acls;
+    c.validations = app.models[c.name].definition.settings.validations;
+    c.isUser = c.sharedClass.ctor.prototype instanceof app.loopback.User ||
+      c.sharedClass.ctor.prototype === app.loopback.User.prototype;
+    if (c.isUser) {
+      return;
+    }
+    result[name] = c;
+  });
+
+  buildScopes(result);
```

Comment:
IIRC, all relations (e.g. "Product belongsTo Category") are implemented as scopes, thus we need scope methods too. 

If you look at [loopback-example-app](https://github.com/strongloop/loopback-example-app), then a scope method is for example `GET /locations/{id}/inventory`, which is `Location.inventory()` a.k.a `Location.__get__inventory()`.

However, feel free to leave them out of this initial pull request if it makes the process easier.